dbx/commands/sync.py

```diff
@@ -94,48 +94,73 @@ def sync():
     pass  # pragma: no cover
 
 
-def create_path_matcher(*, source: str, includes: List[str], excludes: List[str]) -> PathMatcher:
+def create_path_matcher(
+    *,
+    source: str,
+    include_dirs: List[str] = None,
+    exclude_dirs: List[str] = None,
+    include_patterns: List[str] = None,
+    exclude_patterns: List[str] = None,
+    force_include_dirs: List[str] = None,
+    force_include_patterns: List[str] = None,
+    use_gitignore: bool = True,
+) -> PathMatcher:
     """Set up a pattern matcher that is used to ignores changes to files we don't want synced.
 
     Args:
         source (str): root directory includes and excludes are relative to
-        includes (List[str]): Patterns to include.
-        excludes (List[str]): Patterns to exclude.
+        include_dirs (List[str]): Directories to include.
+        exclude_dirs (List[str]): Directories to exclude.
+        include_patterns (List[str]): Patterns to include.
+        exclude_patterns (List[str]): Patterns to exclude.
+        force_include_dirs (List[str]): Directories to include, even if they would otherwise be ignored due to
+                                        exclude dirs/patterns.
+        force_include_patterns (List[str]): Patterns to include, even if they would otherwise be ignored due to
+                                        exclude dirs/patterns.
+        use_gitignore (bool): Whether to use a .gitignore file, if it exists, to populate the list of exclude patterns,
+                              in addition to any other exclude patterns that may be provided.
 
     Returns:
         PathMatcher: matcher for matching files
     """
 
-    ignores = list(DEFAULT_IGNORES)
+    include_dirs = list(include_dirs) if include_dirs else []
+    exclude_dirs = list(exclude_dirs) if exclude_dirs else []
+    include_patterns = list(include_patterns) if include_patterns else []
+    exclude_patterns = list(exclude_patterns) if exclude_patterns else []
+    force_include_dirs = list(force_include_dirs) if force_include_dirs else []
+    force_include_patterns = list(force_include_patterns) if force_include_patterns else []
+
+    include_patterns.extend(subdirs_to_patterns(source, include_dirs))
+    exclude_patterns.extend(subdirs_to_patterns(source, exclude_dirs))
+    force_include_patterns.extend(subdirs_to_patterns(source, force_include_dirs))
 
     gitignore_path = os.path.join(source, ".gitignore")
```

Comment:
@matthayes could you please use [pathlib](https://docs.python.org/3/library/pathlib.html) instead of `os.path` and `with open(..)`?

Reply:
done